api/client/commands.go

```diff
@@ -2660,6 +2666,10 @@ func (s *containerStats) Collect(cli *DockerCli) {
 		for {
 			var v *types.Stats
 			if err := dec.Decode(&v); err != nil {
+				if err == io.EOF && !streamStats {
```

Comment:
Well, I didn't want to change the existing behavior.
And I specifically had to adjust this so that the CLI would actually output data and not an EOF.